compiler/src/dotty/tools/dotc/tastyreflect/ReflectionCompilerInterface.scala

```diff
@@ -117,7 +117,7 @@ class ReflectionCompilerInterface(val rootContext: core.Contexts.Context) extend
   def isInstanceOfPackageClause(using ctx: Context): IsInstanceOf[PackageClause] = new {
     def runtimeClass: Class[?] = classOf[PackageClause]
     override def unapply(x: Any): Option[PackageClause] = x match
```

Comment:
Can we avoid all these unchecked warnings by replacing `x: Any` by `x: tpd.Tree` here and below? /cc @nicolasstucki